### PR TITLE
Attestation metadata update + Improving types on score

### LINKED ISSUE #116: Add OP score to on-chain metadata
- [ ] Add an extra point in metadata for the OP specific score - this involves editing the metadata to be abiencoded instead of just string, and extending it from `string` to `string,uint64`. 
- [ ] Add a version field at the start
the final metadata format is `uint,string,uint`, meaning `version,ipfshash,opscore` respectively

### PR DESCRIPTION
**Pull request by mateodaza**
fixes #116

### CODE REVIEW

**Comment by mateodaza** [comment]
#116

File: src/bulkScore.ts [hunk 1]
```diff
@@ -48,7 +48,7 @@ const processFile = async () => {
 
     const { score, meta } = await createScore(getAddress(address));
     console.log(address, score);
-    const row = [
+    const row: any = [
```
Comment (vacekj) [inline src/bulkScore.ts:51]: no any please 🙂

File: src/helpers/scoreHelpers.ts [hunk 2]
```diff
@@ -1,8 +1,5 @@
 import { Address } from 'viem';
 import {
-  // handleERC20Transactions,
-  // handleERC721Transactions,
-  // handleNormalTransactions,
```
Comment (vacekj) [inline src/helpers/scoreHelpers.ts:5]: same as above - either remove or uncomment if needed.

File: src/helpers/scoreHelpers.ts [hunk 3]
```diff
@@ -20,17 +17,19 @@ import {
...
 export async function createScore(
   address: Address,
-): Promise<{ score: number; meta: any }> {
+): Promise<{ score: number; meta: MetaType }> {
   let score = 0;
-  const meta = {
+  const meta: MetaType = {
     // normalTransactions: null,
     // erc20Transactions: null,
     // erc721Transactions:null,
```
Comment (vacekj) [inline src/helpers/scoreHelpers.ts:29]: ⬆️

File: src/helpers/sourceTypes.ts [hunk 4]
```diff
@@ -1,5 +1,156 @@
...
+  grDonations: {
+    scoreAdded: number;
+    value?: number;
+    applies?: boolean;
+    source: string;
+    category: string;
+    behavior: string;
+  };
+  ethDeposits: {
+    ethDeposits?: any[];
```
Comment (vacekj) [inline src/helpers/sourceTypes.ts:29]: no-any

File: src/helpers/sourceTypes.ts [hunk 5]
```diff
@@ -1,5 +1,156 @@
...
+    scoreAdded: number;
+    applies?: boolean;
+    value?: number;
+    source: string;
+    category: string;
+    behavior: string;
+  };
+  optimismBridges: {
+    opBridges?: any[];
```
Comment (vacekj) [inline src/helpers/sourceTypes.ts:38]: no-any

File: src/helpers/sourceTypes.ts [hunk 6]
```diff
@@ -1,5 +1,156 @@
...
+    scoreAdded: number;
+    applies?: boolean;
+    value?: number;
+    source: string;
+    category: string;
+    behavior: string;
+  };
+  opTreasuryPayouts: {
+    transfers?: any[];
```
Comment (vacekj) [inline src/helpers/sourceTypes.ts:47]: no-any

File: src/helpers/sourceTypes.ts [hunk 7]
```diff
@@ -1,5 +1,156 @@
...
+    isProjectOwner?: boolean;
+    scoreAdded: number;
+    applies?: boolean;
+    value?: boolean;
+    source: string;
+    category: string;
+    behavior: string;
+  };
+  gitcoinPassport: {
+    passport?: any;
```
Comment (vacekj) [inline src/helpers/sourceTypes.ts:113]: no-any

File: src/hooks/useEAS.ts [hunk 8]
```diff
@@ -91,7 +95,12 @@ function useEAS(address: Address | string | Hex | undefined) {
     fetchAttestations();
   }, [address]);
 
-  const mintAttestation = async (score: number, meta: any, scoreData: any) => {
+  const mintAttestation = async (
+    score: number,
+    opScore: number,
+    meta: any,
```
Comment (vacekj) [inline src/hooks/useEAS.ts:101]: no-any

File: src/hooks/useEAS.ts [hunk 9]
```diff
@@ -91,7 +95,12 @@ function useEAS(address: Address | string | Hex | undefined) {
...
+    scoreData: any,
```
Comment (vacekj) [inline src/hooks/useEAS.ts:102]: no-any

**Review by vacekj — CHANGES_REQUESTED** [review]
Great work - there are some `any` left and some commented out code to delete, but otherwise lgtm. I'll aprove asap after those are fixed.

**Comment by mateodaza** [comment]
@vacekj thanks for reviewing, fixes are done :)

**Review by vacekj — APPROVED** [review]

**Pull request merged**